Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -91,32 +101,74 @@ void HiresTexture::Update()
-  };
+  std::vector<std::string> extensions{".dds", ".png", ".bmp",  ".tga",  ".tgb",
+                                      ".jpg", ".jpe", ".jpeg", ".jfif", ".hdr"};
 
-  std::vector<std::string> filenames =
+  std::vector<std::string> Filepaths_Found =
       Common::DoFileSearch({texture_directory}, extensions, /*recursive*/ true);
 
   const std::string code = game_id + "_";
 
-  for (auto& rFilename : filenames)
+  FileType_Counter_DDS_Amount = 0;
+  FileType_Counter_PNG_Amount = 0;
+  FileType_Counter_BMP_Amount = 0;
+  FileType_Counter_TGA_Amount = 0;  // Common for TGA and TGB vatiants
+  FileType_Counter_JPG_Amount = 0;  // Common for JPG, JPE, JPEG and JFIF variants
+  FileType_Counter_HDR_Amount = 0;
+  FileType_Counter_Total_Amount = 0;
+
+  for (auto& rFilepath : Filepaths_Found)
   {
-    std::string FileName;
-    SplitPath(rFilename, nullptr, &FileName, nullptr);
+    std::string CurrentFilename;
+    SplitPath(rFilepath, nullptr, &CurrentFilename, nullptr);
+    SplitPath(rFilepath, nullptr, nullptr, &FileType_Counter_CurrentType);
 
-    if (FileName.substr(0, code.length()) == code)
+    if (CurrentFilename.substr(0, code.length()) == code)
     {
-      s_textureMap[FileName] = rFilename;
+      s_textureMap[CurrentFilename] = rFilepath;
       s_check_native_format = true;
     }
 
-    if (FileName.substr(0, s_format_prefix.length()) == s_format_prefix)
+    if (CurrentFilename.substr(0, s_format_prefix.length()) == s_format_prefix)
     {
-      s_textureMap[FileName] = rFilename;
+      s_textureMap[CurrentFilename] = rFilepath;
       s_check_new_format = true;
     }
+
+    // Check FileType And Count - used by prefetcher
+    if ((FileType_Counter_CurrentType == ".dds") || (FileType_Counter_CurrentType == ".DDS"))
+    {
+      FileType_Counter_DDS_Amount++;
+    }
+
+    if ((FileType_Counter_CurrentType == ".png") || (FileType_Counter_CurrentType == ".PNG"))
+    {
+      FileType_Counter_PNG_Amount++;
+    }
+
+    if ((FileType_Counter_CurrentType == ".bmp") || (FileType_Counter_CurrentType == ".BMP"))
+    {
+      FileType_Counter_BMP_Amount++;
+    }
+
+    if ((FileType_Counter_CurrentType == ".tga") || (FileType_Counter_CurrentType == ".TGA") ||
+        (FileType_Counter_CurrentType == ".tgb") || (FileType_Counter_CurrentType == ".TGB"))
+    {
+      FileType_Counter_TGA_Amount++;
+    }
+
+    if ((FileType_Counter_CurrentType == ".jpg") || (FileType_Counter_CurrentType == ".JPG") ||
+        (FileType_Counter_CurrentType == ".jpe") || (FileType_Counter_CurrentType == ".JPE") ||
+        (FileType_Counter_CurrentType == ".jpeg") || (FileType_Counter_CurrentType == ".JPEG") ||
+        (FileType_Counter_CurrentType == ".jfif") || (FileType_Counter_CurrentType == ".JFIF"))
+    {
+      FileType_Counter_JPG_Amount++;
+    }
+
+    if ((FileType_Counter_CurrentType == ".hdr") || (FileType_Counter_CurrentType == ".HDR"))
+    {
+      FileType_Counter_HDR_Amount++;
+    }
```

Comment:
Will do that now and see how clang-format does it, but this wouldn't apply if a map is used?